App.tsx

```diff
@@ -130,90 +131,106 @@ const App = () => {
-    });
-
-    const onLayoutRootView = useCallback(async () => {
-      if (fontsLoaded) {
-        await SplashScreen.hideAsync();
-      }
-    }, [fontsLoaded]);
-
-    if (!fontsLoaded) {
-      // The splash screen keeps rendering while fonts are loading
-      return null;
-    }
-
     return (
-      <ClientContext.Provider value={contextValue}>
-        <QueryClientProvider client={queryClient}>
-          <NativeBaseProvider theme={theme}>
-            <SafeAreaProvider>
-              <NavigationContainer>
-                <View onLayout={onLayoutRootView} style={StyleSheet.absoluteFill}>
-                  <TabNavigator.Navigator
-                    initialRouteName="Home"
-                    screenOptions={({route}) => ({
-                      headerShown: false,
-                      tabBarIcon: ({color, size}) => {
-                        if (route.name === 'Home') {
-                          return <AntDesign name="search1" size={size} color={color} />;
-                        } else if (route.name === 'Observations') {
-                          return <AntDesign name="filetext1" size={size} color={color} />;
-                        } else if (route.name === 'Weather Data') {
-                          return <AntDesign name="barschart" size={size} color={color} />;
-                        } else if (route.name === 'Menu') {
-                          return <AntDesign name="bars" size={size} color={color} />;
-                        }
-                      },
-                    })}>
-                    <TabNavigator.Screen name="Home" initialParams={{center_id: avalancheCenterId, date: date}}>
-                      {state => HomeTabScreen(withParams(state, {center_id: avalancheCenterId, date: date}))}
-                    </TabNavigator.Screen>
-                    <TabNavigator.Screen name="Observations" initialParams={{center_id: avalancheCenterId, date: date}}>
-                      {state => ObservationsTabScreen(withParams(state, {center_id: avalancheCenterId, date: date}))}
-                    </TabNavigator.Screen>
-                    <TabNavigator.Screen name="Weather Data" initialParams={{center_id: avalancheCenterId, date: date}}>
-                      {state => TelemetryTabScreen(withParams(state, {center_id: avalancheCenterId, date: date}))}
-                    </TabNavigator.Screen>
-                    <TabNavigator.Screen name="Menu" initialParams={{center_id: avalancheCenterId}}>
-                      {() => MenuStackScreen(avalancheCenterId, setAvalancheCenterId, staging, setStaging)}
-                    </TabNavigator.Screen>
-                  </TabNavigator.Navigator>
-                </View>
-              </NavigationContainer>
-            </SafeAreaProvider>
-          </NativeBaseProvider>
-        </QueryClientProvider>
-      </ClientContext.Provider>
+      <QueryClientProvider client={queryClient}>
+        <AppUsingQueryClient />
+      </QueryClientProvider>
```

Comment:
this refactoring doesn't make the code immediately clearer to me - the old structure was deeply nested to be sure, but at least it was clear.

I could see an argument for splitting it up if it was to get a particular chunk of the component under test, but this feels kind of arbitrary.

Reply:
It was just because I needed to get the hook to still be as top-level as possible, but within a component that had the query context - what would you suggest instead?